### DISCUSSION TITLE
How to specify which surfaces to include in aerostructural analysis?

### DISCUSSION THREAD

**Discussion by arnalj** [mdolab/dafoam#470] [General]
I am interested in setting up an aerostructural optimization case where only one surface is structurally deformed, however I'm not confident on how I would do this. For example, imagine a case with two wall patches: `'body'` and `'wing'`. How do I specify that only `'wing'` should undergo structural analysis and deformation? Would it be like this?:

```
daOptions = {
    "designSurfaces": ["wing", "body"],
    "solverName": "DASimpleFoam",
    "couplingInfo": {
        "aerostructural": {
                "active": False,
                "pRef": 0,
                "propMovement": False,
                "couplingSurfaceGroups": {
                    "wingGroup": ["wing"],
                },
            },
        },
     .
     .
     .
   }
```

Will MELD know to only do the displacement and load transfer on `'wing'` nodes?

**Comment by friedenhe** [comment]
The above code looks right and will do what you want to. Make sure to change the "active" from "False" to "True"

**Comment by arnalj** [comment]
Thank you Professor

**Reply by NUAA-qinfei** [reply]
Hi,I have the same question,I want to know if you success to do aerostructural opt  with two wall patches: 'body' and 'wing'?

**Comment by friedenhe** [comment]
Yes, it should work for two wall patches. But you need to use the latest DAFoam version. We used to have a bug if more than one patch was used.

**Reply by NUAA-qinfei** [reply]
thank you for prof's replay,I have to wall patch, "wing" and "nacelle", and have two FFD blocks,one FFD controls the "wing"shape, another FFD controls the  "nacelle" position, and the "wing" patch have struct mesh , but "nacelle" patch no struct, I konw for the "wing" opt should use  "scenario_aerostructural" ,but for the "nacelle" just the "scenario_areodynamic",I want to know if DAFoam can support this opt. (if no wing struct,I do it successfully,but know,I want to couple the wing struct to consider the wing static aeroelasticity)
 
![image](https://github.com/user-attachments/assets/053f19a5-a8c4-4a9b-b59f-656209fdfa7d)
![image](https://github.com/user-attachments/assets/e63cc9e0-565f-43f3-a388-5d2758d75e20)
![image](https://github.com/user-attachments/assets/42c07c87-773f-4b8c-964f-47c6a7bfb09b)
